source/data-formats/bson.txt

```diff
@@ -212,6 +216,41 @@ BSON document stored in ``myFile.bson``:
    ``System.IO.Stream`` object. This means that you can read or write any location 
    that can be accessed by a stream.
 
+Implement a Custom BSON Reader and Writer
+-----------------------------------------
+
+The preceding examples show how to read and write BSON documents by using the
+``BsonBinaryReader`` and ``BsonBinaryWriter`` classes. However, you can also implement a
+custom reader and writer by implementing the
+``IBsonReader`` and ``IBsonWriter`` interfaces. This is useful when you want to define
+a custom serializer or if you want to directly read or write BSON or JSON data without
```

Comment:
While writing a custom serializer is an expected (if rare) thing for a user to do, there is no need for custom readers and writers.

Custom serializers use the standard reader and writer interfaces to `Deserialize` and `Serialize` values.